proto/dictation_asr.proto

```diff
@@ -418,46 +506,110 @@ message StreamingRecognizeResponse {
     // until the server closes the gRPC connection. This event is only sent if
     // `single_utterance` was set to `true`, and is not used otherwise.
     END_OF_SINGLE_UTTERANCE = 1;
-    
+
+    // [**Extension by Techmo**]
+    // This event indicates that recognition has started and there was no speech detected
+    // for a certain period of time. The server will not process
+    // additional audio (although it may subsequently return additional results).
+    // The client should stop sending additional audio data, half-close the gRPC
+    // connection, and wait for any additional results until the server closes
+    // the gRPC connection.
+    // MRCPv2-related: [No-Input-Timeout](https://tools.ietf.org/html/rfc6787#section-9.4.6)
+    NO_INPUT_TIMEOUT = 2;
+
     // [**Extension by Techmo**]
-    // This event indicates that the server has detected no speech input timeout
-    // and expects no additional speech. Therefore, the server will not process
+    // This event indicates that recognition has started and there was no result
+    // for a certain period of time. The server will not process
     // additional audio (although it may subsequently return additional results).
     // The client should stop sending additional audio data, half-close the gRPC
     // connection, and wait for any additional results until the server closes
     // the gRPC connection.
-    NO_INPUT_TIMEOUT = 4;
+    // MRCPv2-related: [Recognition-Timeout](https://tools.ietf.org/html/rfc6787#section-9.4.7)
+    RECOGNITION_TIMEOUT = 3;
+
+    // [**Extension by Techmo**]
+    // [Speech-Complete-Timeout](https://tools.ietf.org/html/rfc6787#section-9.4.15)
+    // This event indicates that recognition has been finalized with a complete result
+    // after specified length of silence after user speech.
+    SPEECH_COMPLETE_TIMEOUT = 4;
```

Comment:
Remove TIMEOUT, this is standard case. Do we assume that from now we're returning Event type for every message ?

Reply:
It looks like we do. For every result, an event indicating recognition termination cause will be sent first. Do you see any potential drawback of this approach?

Reply:
Can we simplify communication flow and return both event type and results (when applicable) in single message?

Reply:
I would say no. From GSAPI:
```cpp
// - In each response, only one of these fields will be set:
//     `error`,
//     `speech_event_type`, or
//     one or more (repeated) `results`.
```